sdk/metric/view/view_test.go

```diff
@@ -0,0 +1,348 @@
+			notMatch: emptyDescription,
+		},
+		{
+			name: "Library version",
+			options: []Option{
+				MatchInstrumentationLibrary(instrumentation.Library{
+					Version: "v1.0.0",
+				}),
+			},
+
+			match:    matchInstrument,
+			notMatch: emptyDescription,
+		},
+		{
+			name: "Library SchemaURL",
+			options: []Option{
+				MatchInstrumentationLibrary(instrumentation.Library{
+					SchemaURL: "stuff.test/",
+				}),
+			},
+			match:    matchInstrument,
+			notMatch: emptyDescription,
+		},
+		{
+			name: "Expands *",
+			options: []Option{
+				MatchInstrumentName("f*"),
+			},
+			match:    matchInstrument,
+			notMatch: emptyDescription,
+		},
+		{
+			name: "composite literal name",
+			options: []Option{
+				MatchInstrumentName("foo"),
+				MatchInstrumentationLibrary(instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				}),
+			},
+			match:    matchInstrument,
+			notMatch: emptyDescription,
+		},
+
+		{
+			name: "rename",
+			options: []Option{
+				MatchInstrumentName("foo"),
+				WithName("baz"),
+			},
+			match: Instrument{
+				Scope: instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				},
+				InstrumentName:        "baz",
+				InstrumentDescription: "",
+			},
+			notMatch: emptyDescription,
+		},
+		{
+			name: "rename",
+			options: []Option{
+				MatchInstrumentName("foo"),
+				WithDescription("descriptive stuff"),
+			},
+			match: Instrument{
+				Scope: instrumentation.Library{
+					Name:      "bar",
+					Version:   "v1.0.0",
+					SchemaURL: "stuff.test/",
+				},
+				InstrumentName:        "foo",
+				InstrumentDescription: "descriptive stuff",
+			},
+			notMatch: emptyDescription,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v, err := New(tt.options...)
+			require.NoError(t, err)
+
+			t.Run("match", func(t *testing.T) {
+				got, match := v.TransformInstrument(matchInstrument)
+				assert.Equal(t, tt.match, got)
+				assert.True(t, match)
+			})
+
+			t.Run("does not match", func(t *testing.T) {
+				got, match := v.TransformInstrument(noMatchInstrument)
+				assert.Equal(t, tt.notMatch, got)
+				assert.False(t, match)
+			})
+		})
+	}
+}
+
+func TestView_matchName(t *testing.T) {
+	tests := []struct {
+		name        string
+		matchName   string
+		matches     []string
+		notMatches  []string
+		hasWildcard bool
+	}{
+		{
+			name:        "exact",
+			matchName:   "foo",
+			matches:     []string{"foo"},
+			notMatches:  []string{"foobar", "barfoo", "barfoobaz"},
+			hasWildcard: false,
+		},
+		{
+			name:        "*",
+			matchName:   "*",
+			matches:     []string{"foo", "foobar", "barfoo", "barfoobaz"},
+			notMatches:  []string{},
+			hasWildcard: true,
+		},
+		{
+			name:        "front ?",
+			matchName:   "?foo",
+			matches:     []string{"1foo", "afoo"},
+			notMatches:  []string{"foo", "foobar", "barfoo", "barfoobaz"},
```

Comment:
`"foo"` should match the wildcard `?foo`

```suggestion
			matches:     []string{"1foo", "afoo", "foo"},
			notMatches:  []string{"foobar", "barfoo", "barfoobaz"},
```

Reply:
That is an exactly one match.

> with the question mark (?) matching exactly one character